service_api.md

```diff
@@ -0,0 +1,284 @@
+# Service API
+
+The idea of the service API is to create a uniform interface to interact with Data Manager
+ressources from a user agent, for example the browser. Since the project uses Vaadin, the API is
+written in Java and provides an easy integration
+in Vaadin Flow.
+
+The API provides access and manipulation of resources similar to what you know from REST, with
+slightly different names though:
+
+- Create (similar to POST)
+- Get (similar to GET)
+- Update (similar to PATCH, partial updates are possible)
+- Delete (similar to DELETE)
+
+This documentation describes the conceptual idea of how the service API is composed and works.
+Details and semantics
+are suspected to change.
+
+## Create examples
+
+Creating a new resource shall be intuitive and straight forward. No new concepts than the resource
+model
+itself should be required.
+For example, to create a new project, just create a `Project Information` object without an ID (
+there is none, needs to be given by the system).
+Once the resource has been created, the requests will result in the same object model with an ID.
+
+To provide clients with an easy-to-use resource creation interface, the API shall return
+the created resource under the same model again, provided with the generated ID.
+
+### Create a project
+
+```mermaid
+---
+title: Create Project Example
+---
+classDiagram
+    class AsyncProjectService {
+        Mono~ProjectCreationResponse~ create(ProjectCreationRequest request)
+    }
+    <<interface>> AsyncProjectService
+    AsyncProjectService ..> ProjectCreationRequest
+    AsyncProjectService ..> ProjectCreationResponse
+
+    class ProjectCreationRequest {
+        String projectId
+        ProjectInformation requestBody
+        String requestId
+    }
+
+    class ProjectCreationResponse {
+        String projectId
+        ProjectInformation responseBody
+        String requestId
+    }
+
+    class ProjectInformation
+
+    ProjectCreationRequest ..> ProjectInformation
+    ProjectCreationResponse ..> ProjectInformation
+```
+
+### Creating a sub-resource
+Creating a sub-resource that are in a many-to-one relationship with their parent resource should be made easy.
+
+For example, creating an `Example Group` for an existing `Experiment` shall be a no-brainer. Therefore, it is 
```

Comment:
I think this is `Experimental Group` instead of `Example Group`. If you intended to write `Example Group` here, it should be `Example Group` in the create method as well.